### ISSUE TITLE
[Enhancement] Enable HTTP auth header to be configurable.

### ISSUE THREAD

**Issue by dsimmons** [sahat/satellizer#169]
[In the current version](https://github.com/sahat/satellizer/blob/0.8.1/satellizer.js#L602), the assumption (and a reasonable one at that) is that your backend uses the HTTP "standard" Authorization header.

``` Javascript
if (localStorage.getItem(tokenName)) {
  httpConfig.headers.Authorization = 'Bearer ' + localStorage.getItem(tokenName);
}
```

As I mentioned, this is a very reasonable assumption, but I think it makes sense to allow this to be a configuration option (with the above being the default). 
#### Use case

My backend expects a custom `x-access-token` header. Although the `POST` for the JWT at login would occur successfully, the behavior I was seeing was that the token would disappear within a split-second.

It was due to the following:

``` Javascript
responseError: function(response) {
  // TODO: check if coming from the same origin or pre-approved domain
  if (response.status === 401) {
    localStorage.removeItem(tokenName);
  }
  return $q.reject(response);
}
```

Basically, the front-end would see that some of my resource requests beyond login were failing with an `HTTP 401` due to the fact that the auth header was "not present" (not via the expected header key anyway), and delete the token.  Therefore, my "login" would last until a full-page refresh, then throw me back to the login screen.

That said, I'm happy to submit a PR and could have it done in short order, just wanted to make sure it made sense before I spent time doing so! :smile:
